### PR TITLE
Feature data assimilation2

### LINKED ISSUE #204: Optimisation of DA
- output as binary
- remove `getline` to skip lines, use `seekg` (line length is known due to binary format)
- reserve string length in `create_header` (length is calculable)
-

### PR DESCRIPTION
**Pull request by ChristianVonMach**
closes #204 

instead of using a custom fileformat -> uses hdf5

verified with example in `data_assimilation`

image

image

### CODE REVIEW

**Comment by ChristianVonMach** [comment]
`visualisation.dat` example structure
```
/
-metatdata (Group)
  |- dt (Field)
  |-domain
  ...
-T_0.1 (Group)
  |- u    (Field)
  |- v
  |- w
...
-T_0.2
  |- u
...
-u (input from python script)
-v (input from python script)
...
```

File: src/dataAssimilation/FieldIO.cpp [hunk 1]
```diff
@@ -46,30 +44,24 @@ FieldIO::FieldIO(const std::string &xml_file_name, const std::string &output_fil
...
+    HighFive::File out_file(m_file_name, HighFive::File::ReadWrite);
+    auto group_name = fmt::format("{:.5e}", t_cur);
+    m_logger->debug("attempt to write @t:{}", t_cur);
+
+    if (out_file.exist(group_name)) {
+        out_file.unlink(group_name);
+    }
+    
+    HighFive::Group t_group = out_file.createGroup(group_name);
+    // Field fields[] = {u, v, w, p, T, C};
```
Comment (LinhWuerzburger) [inline src/dataAssimilation/FieldIO.cpp:56]: old code?

File: src/dataAssimilation/FieldIO.cpp [hunk 2]
```diff
@@ -85,22 +77,14 @@ void FieldIO::write_fields(real t_cur, Field &u, Field &v, Field &w, Field &p, F
...
-    read_field(input_file, w);
-    read_field(input_file, p);
-    read_field(input_file, T);
-    read_field(input_file, C);
+    HighFive::File input_file(m_file_name, HighFive::File::ReadOnly);
+
+    read_vis_field(input_file, u, t_cur);
+    read_vis_field(input_file, v, t_cur);
+    read_vis_field(input_file, w, t_cur);
+    // read_vis_field(input_file, p, t_cur);
```
Comment (LinhWuerzburger) [inline src/dataAssimilation/FieldIO.cpp:85]: forgot to remove the comment?

File: src/dataAssimilation/FieldIO.cpp [hunk 3]
```diff
@@ -111,49 +95,73 @@ void FieldIO::read_fields(real t_cur, Field &u, Field &v, Field &w, Field &p, Fi
...
+    auto full_name = fmt::format("/{:.5e}/{}", t, field_name);
+    m_logger->info("opening vis file at {}", full_name);
+    auto ds = file.getDataSet(full_name);
+    ds.read(field.data);
 }
 
 void FieldIO::read_fields(const Settings::data_assimilation::field_changes &field_changes,
                           Field &u, Field &v, Field &w,
                           Field &p, Field &T, Field &C) {
+    int n;
```
Comment (LinhWuerzburger) [inline src/dataAssimilation/FieldIO.cpp:151]: unused?

Comment (ChristianVonMach) [reply src/dataAssimilation/FieldIO.cpp]: done in #7129f6ac

File: src/dataAssimilation/FieldIO.cpp (hunk 1 above)
Comment (ChristianVonMach) [reply src/dataAssimilation/FieldIO.cpp]: done in 3065b6bb

File: src/dataAssimilation/FieldIO.cpp (hunk 2 above)
Comment (ChristianVonMach) [reply src/dataAssimilation/FieldIO.cpp]: done in 3065b6bb

**Pull request closed without merging**